Get rid of unused wxTransformMatrix

a-wi opened the pull request:
As it is stated in matrix.cpp wxTransformMatrix was intended to be used
in wxDC to replace the basic system of scaling/translation but actually
it was never used.
Because applying affine transformations can be done with wxAffineMatrix2D
and wxDC::SetTransformMatrix() we can get rid of this dead code.

vadz commented on build/bakefiles/files.bkl:
```diff
@@ -883,7 +883,6 @@ IMPORTANT: please read docs/tech/tn0016.txt before modifying this file!
     src/common/lboxcmn.cpp
     src/common/listctrlcmn.cpp
     src/common/markupparser.cpp
-    src/common/matrix.cpp
```
This file isn't supposed to be modified directly, please update `build/files` and run `./build/upmake` instead.

vadz reviewed (approved):
Thanks for cleaning this up, I indeed have no idea why did we have it for so long when it was never used.

catalinr commented:
There is also [ticket 13114](http://trac.wxwidgets.org/ticket/13114) that should be closed when this is merged.

a-wi committed:
Get rid of unused wxTransformMatrix

As it is stated in matrix.cpp wxTransformMatrix was intended to be used
in wxDC to replace the basic system of scaling/translation but actually
it was never used.
Because applying affine transformations can be done with wxAffineMatrix2D
and wxDC::SetTransformMatrix() we can get rid of this dead code.

Closes #13114.
